tools/fragpipe/fragpipe.xml

```diff
@@ -0,0 +1,286 @@
+<tool id="fragpipe" name="FragPipe -  Academic Research and Education User License (Non-Commercial)" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" python_template_version="3.5">
+    <description>Data analysis for mass spectrometry-based proteomics</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+
+    <stdio>
+        <!-- The error code from individual tools run by FragPipe is not always propogated to the FragPipe process exit code -->
+        <regex match="exit code: [^0]"
+               source="stdout"
+               level="fatal"
+               description="A FragPipe pipeline process returned non-zero exit code."/>
+    </stdio>
+
+    <version_command>fragpipe --help | grep 'FragPipe v'</version_command>
+
+    <!--
+        FragPipe wrapper command:
+        - Links input files
+        - Modifies database path in workflow file
+        - Runs Philosopher to add decoys and contaminants
+        - Runs FragPipe, passing paths to MSFragger, Philosopher, Python, and IonQuant
+    -->
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir .cache .config tmp &&
+        export XDG_CONFIG_HOME=\$(pwd)/.config &&
+        export XDG_CACHE_HOME=\$(pwd)/.cache &&
+        export JAVA_OPTS="\$JAVA_OPTS -Xmx\$[ \${GALAXY_MEMORY_MB:-8192} / 1024 ]g -Djava.io.tmpdir=\$(pwd)/tmp" &&
```

Comment:
> Not sure if setting the -Xmx in the tool is a good idea.

Ah, yes. This should already be handled via via the `fragpipe --ram` flag. I should have removed this.

> Also overwriting JAVA_OPTS seems not not be a good idea.

This was the action suggested by the developer to specify a non-system tmp location: https://github.com/Nesvilab/FragPipe/issues/478. That said, it looks like there may be a bug in fragpipe such that it's not being correctly used.

Is there a specific reason not to supplement `JAVA_OPTS` in the tool xml?

Reply:
My mistake. Since you are not overwriting, but appending to the variable this  is fine.

Reply:
Cool. I removed the `-Xmx`.

Reply:
I also addressed the local cache issue I mentioned with this commit (https://github.com/galaxyproteomics/tools-galaxyp/pull/761/commits/3031e8f28e182197f68151537bee32f81616c61b).

You can see here (https://github.com/Nesvilab/FragPipe/blob/develop/MSFragger-GUI/src/com/dmtavt/fragpipe/FragpipeLocations.java#L91) that FragPipe always resolves its cache location relative to its jar location, so I copy the jar to the tool directory so that it can create the cache.

The alternative is that on a system where the user running the Galaxy job has read/write access to the conda installation, there could be multiple FragPipe processes using the same cache.